### PR TITLE
Adding jarm fingerprinting

### PR DESCRIPTION
**Pull request by Mzack9999**
## Description
This PR implements jarm fingerprinting

## Notes
`ja3` and `ja3s` will require two independent follow-up tickets as it's necessary to access client hello and server hello raw bytes of the tls/ztls client. I suggest to grab the specific hello packets by intercepting TCP traffic with gopacket with an eBpf filter for specific TLS steps, in this way it's fully passive without any additional TLS connection required.

## Example
```console
$ echo hackerone.com:443 | go run . -jarm -silent
hackerone.com:443 [29d3dd00029d29d00042d43d00041d5de67cc9954cc85372523050f20b5007]
```

### CODE REVIEW

**Comment by ehsandeep** [comment]
Follow up tickets: 
- https://github.com/projectdiscovery/tlsx/issues/41
- https://github.com/projectdiscovery/tlsx/issues/42

**Review by ehsandeep — APPROVED** [review]
@Mzack9999

Just to confirm, connection pooling is considered for this implementation as mentioned in https://github.com/projectdiscovery/httpx/issues/576?

File: pkg/tlsx/jarm/regex.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package jarm
+
+import "regexp"
+
+var (
+	JarmHashRegex = regexp.MustCompile("(?m)0{62}")
```
Comment (Ice3man543) [inline pkg/tlsx/jarm/regex.go:6]: why do we need this regex? this seems like unncessary

Comment (Mzack9999) [reply pkg/tlsx/jarm/regex.go]: true, checked the library code and it's already returning a sha256 hex encoded

File: pkg/tlsx/jarm/jarm.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+		_, err = c.Write(gojarm.BuildProbe(probe))
+		if err != nil {
+			results = append(results, "")
+			c.Close()
+			continue
+		}
+		_ = c.SetReadDeadline(time.Now().Add(timeout))
+		buff := make([]byte, 1484)
+		_, _ = c.Read(buff)
+		c.Close()
```
[the remark this answers is not in the document]
Comment (Mzack9999) [reply pkg/tlsx/jarm/jarm.go]: fixed

**Pull request merged**